src/aggregate/aggregate_request.c

```diff
@@ -1058,29 +1053,23 @@ static ResultProcessor *getArrangeRP(AREQ *req, AGGPlan *pln, const PLN_BaseStep
       // add it to the loadkeys list.
       if(!(sortkey->flags & RLOOKUP_F_SVSRC) &&
          !(sortkey->flags & RLOOKUP_F_ISLOADED)) {
-        // initialize loadKeys
-        if (!loadKeys) {
-          loadKeys = rm_calloc(nkeys, sizeof(*loadKeys));
-        }
-        // Add to loadkeys
-        loadKeys[nLoadKeys++] = sortkey;
+
+        *array_ensure_tail(&loadKeys, const RLookupKey *) = sortkey;
 
         // Update lookupkey's flag to loaded
         sortkey->flags |= RLOOKUP_F_ISLOADED;
       }
     }
-
-    rp = RPSorter_NewByFields(limit, sortkeys, nkeys, astp->sortAscMap);
-
     if(loadKeys) {

```

Comment:
is there a reason to free the `loadkeys` array if we already collected and allocated it? why not move it as is to the constructor?